src/shapes/sumOfLorentzians.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * This function calculates the spectrum as a sum of lorentzian functions. The Lorentzian
+ * parameters are divided in 3 batches. 1st: centers; 2nd: heights; 3th: widths;
+ * @param t Ordinate values
+ * @param p Lorentzian parameters
+ * @returns {*}
+ */
+
+export function sumOfLorentzians(p) {
+  return function (t) {
+    let nL = p.length / 3;
+    let result = 0;
+    for (let i = 0; i < nL; i++) {
+      let p2 = Math.pow(p[i + nL * 2] / 2, 2);
+      let factor = p[i + nL] * p2;
+      result += factor / (Math.pow(t - p[i], 2) + p2);
```

Comment:
The result should be multiplied by 1 / Math.PI? Normalization purpose

Reply:
For this case, we want to find the fwhm, intensity, and position of each lorentzian of the spectrum so these Lorentzian are not normalized. Is it that you mean?